homeassistant/components/home_plus_control/switch.py

```diff
@@ -0,0 +1,139 @@
+"""Legrand Home+ Control Switch Entity Module that uses the HomeAssistant DataUpdateCoordinator."""
+from functools import partial
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_OUTLET,
+    DEVICE_CLASS_SWITCH,
+    SwitchEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import dispatcher
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DATA_COORDINATOR,
+    DISPATCHER_REMOVERS,
+    DOMAIN,
+    HW_TYPE,
+    SIGNAL_ADD_ENTITIES,
+)
+
+
+@callback
+def add_switch_entities(new_unique_ids, coordinator, add_entities):
+    """Add switch entities to the platform.
+
+    Args:
+        new_unique_ids (set): Unique identifiers of entities to be added to Home Assistant.
+        coordinator (DataUpdateCoordinator): Data coordinator of this platform.
+        add_entities (function): Method called to add entities to Home Assistant.
+    """
+    new_entities = []
+    for uid in new_unique_ids:
+        new_ent = HomeControlSwitchEntity(coordinator, uid)
+        new_entities.append(new_ent)
+    add_entities(new_entities)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Legrand Home+ Control Switch platform in HomeAssistant.
+
+    Args:
+        hass (HomeAssistant): HomeAssistant core object.
+        config_entry (ConfigEntry): ConfigEntry object that configures this platform.
+        async_add_entities (function): Function called to add entities of this platform.
+    """
+    partial_add_switch_entities = partial(
+        add_switch_entities, add_entities=async_add_entities
+    )
+    # Connect the dispatcher for the switch platform
+    hass.data[DOMAIN][config_entry.entry_id].get(DISPATCHER_REMOVERS).append(
+        dispatcher.async_dispatcher_connect(
+            hass, SIGNAL_ADD_ENTITIES, partial_add_switch_entities
+        )
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    coordinator = hass.data[DOMAIN][config_entry.entry_id].get(DATA_COORDINATOR)
+    await coordinator.async_refresh()
```

Comment:
We can move this to the init module if we make sure that the platform is loaded first.
```suggestion
```